lib/miq_expression.rb

```diff
@@ -1370,10 +1365,10 @@ def fields(expression = exp)
 
       if (val = expression["field"] || expression["count"] || expression["tag"])
         ret = []
-        tg = self.class.parse_field_or_tag(val)
-        ret << tg if tg
-        tg = self.class.parse_field_or_tag(expression["value"].to_s)
-        ret << tg if tg
+        tg = Target.parse(val)
+        ret << tg unless tg.kind_of?(InvalidTarget)
```

Comment:
Should this be something like

```suggestion
        ret << tg if tg.valid?
```

Reply:
Discussed this w/ Jason

I had originally intended on making this `tg.valid?`, but `.valid?` is a little more stringent.

And the future play is to drop this parsing, so I didn't want to have future code question if we need to add a `valid?` check.